sorting/gnome_sort.cpp

```diff
@@ -0,0 +1,29 @@
+#include <iostream>
+
+void gnomesort(int arr[], int size) {
```

Comment:
It would be ideal if you can template this function somethin along the lines of:
```cpp
template<typename T>
void gnomesort(T arr[]) {
```
```cpp
template<typename T>
void gnomesort(std::vector<T> *arr) {
std::size_t size = arr->size()

// access elements of arr -> 
std::swap((*arr)[index], (*arr)[index - 1]);  // swap

```

Reply:
It would be but..
I checked every file in sort folder and other folders and none of them has it. So i think at this point its not important to do it for merging. We can create issue after merging it about every other file translating to Template. Also, there are some style errors in other files and std ones so i think we have to create same issue about them and change their style.

Reply:
In summary, i dont think i have to write it on this issue i can create new one for it.

Reply:
We are in the process of standardizing and improving the quality of the code. see the quality of code being updated to -here - [code sample](https://github.com/kvedala/C-Plus-Plus/blob/master/machine_learning/kohonen_som_trace.cpp) and its auto-generated [documentation](https://kvedala.github.io/C-Plus-Plus/d9/d49/kohonen__som__trace_8cpp.html)
Once a problem is identified, it is kinda pointless to repeat the same errors. 
Hence, we are ensuring that the new pull requests are meeting the standards and actually add an educational value and that the repo is not a dump your code sort of deal. 
Thank you

Reply:
Got it. The main reason why i wanted new issue was to resolve other files too. I will fix it but we have to create new issue regarding that problem also. Dont you agree?

Reply:
we can continue with this PR - that way tracking the changes is much easier thus voiding re-reading the code again. hence, there would not be a need for another pull. - unless you may want to have a branch and pull on your fork independently of this repo :)

Reply:
Deal!

Reply:
> Deal! Will get to work right away. Thanks for taking time and review my code.

absolutely. thank you for contributing 👍